omnigan/utils.py

```diff
@@ -291,46 +292,52 @@ def make_json_file(
     '/network/tmp1/ccai/data/munit_dataset/trainA_size_1200/',
     '/network/tmp1/ccai/data/munit_dataset/seg_trainA_size_1200/',
     '/network/tmp1/ccai/data/munit_dataset/trainA_megadepth_resized/'
-    ], 'train_r_resized.json')
+    ], ["train_r.json", "val_r.json"])
 
     Args:
-        keys (list): the list of image type like 'x', 'm', 'd', etc.
-        addresses (list): the list of the corresponding address of the image type mentioned in keys.
-        name_of_the_json_file (str, optional): The name of the output json file. Default to "jsonfile.json"
+        tasks (list): the list of image type like 'x', 'm', 'd', etc.
+        addresses (list): the list of the corresponding address of the image type mentioned in tasks
+        json_names (list): names for the json files, train being first (e.g. : ["train_r.json", "val_r.json"])
         splitter (str, optional): The path separator for the current OS. Defaults to '/'.
+        pourcentage_val: pourcentage of files to go in validation set
     """
+    assert len(tasks) == len(addresses), "keys and addresses must have the same length!"
 
-    print("Please Make sure there is a file with the same name in each folder!")
-    assert len(keys) == len(addresses), "keys and addresses must have the same length!"
-
-    files = [get_files(addresses[j]) for j in range(len(keys))]
+    files = [get_files(addresses[j]) for j in range(len(tasks))]
```

Comment:
I think ideally I wouldnt trust that there are the same number of files, but I would check that the basenames match, and check that the paths exists etc. but I guess that's good enough for our purpose . (eg I once made a file where the train and test were not actually in the correct folders, so mixed up in between folders, so that wouldn't hae worked)

Reply:
Ahhh yeah I agree with you. I mainly modified the function to include the possibility to divide between train and val. But I didn't really review the base code. But I guess, like you said, that it's good enough for our purpose.